### PR TITLE
run containerized mode without host mode enabled

### PR DESCRIPTION
**Pull request by Ankurk99**
Currently KubeArmor when run only in containerized mode (without host mode) -enableKubeArmorPolicy is not receiving policies due to dependency on -enableKubeArmorHostPolicy. This fix will enable us to run containerized mode without -enableKubeArmorHostPolicy

### CODE REVIEW

File: KubeArmor/core/kubeArmor.go [hunk 1]
```diff
@@ -616,7 +616,7 @@ func KubeArmor() {
 		dm.Logger.Print("Started to monitor container security policies on gRPC")
 	}
 
-	if !cfg.GlobalCfg.K8sEnv && cfg.GlobalCfg.HostPolicy {
+	if !dm.K8sEnabled {
```
Comment (daemon1024) [inline KubeArmor/core/kubeArmor.go]: Wouldn't this enable Host Policy even if HostPolicy is disabled?

Comment (Ankurk99) [reply KubeArmor/core/kubeArmor.go]: No, we will receive the host policy but it won't be enforced.

Comment (daemon1024) [reply KubeArmor/core/kubeArmor.go]: But we shouldn't be receiving the host policy :)

Comment (Ankurk99) [reply KubeArmor/core/kubeArmor.go]: Actually we are checking again if hostPolicy is enabled when creating the apparmor profile and applying it. Here: https://github.com/kubearmor/KubeArmor/blob/main/KubeArmor/enforcer/appArmorEnforcer.go#L588

**Comment by daemon1024** [comment]
Idk y did this close 😓 Follow up #852

**Pull request merged**